proxy/proxy.go

```diff
@@ -3,82 +3,90 @@ package proxy
 import (
 	"fmt"
 	"net/url"
-	"time"
+	"os"
+	"os/signal"
+	"context"
+	"syscall"
 
-	"github.com/9seconds/httransform"
+	"github.com/9seconds/httransform/v2"
+	"github.com/9seconds/httransform/v2/auth"
+	"github.com/9seconds/httransform/v2/layers"
+	"github.com/9seconds/httransform/v2/dialers"
+	"github.com/9seconds/httransform/v2/executor"
 
 	"github.com/scrapinghub/crawlera-headless-proxy/config"
-	"github.com/scrapinghub/crawlera-headless-proxy/layers"
+	customs "github.com/scrapinghub/crawlera-headless-proxy/layers"
 	"github.com/scrapinghub/crawlera-headless-proxy/stats"
 )
 
 func NewProxy(conf *config.Config, statsContainer *stats.Stats) (*httransform.Server, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		for range signals {
+			cancel()
+		}
+	}()
+
 	crawleraURL, err := url.Parse(conf.CrawleraURL())
 	if err != nil {
 		return nil, fmt.Errorf("incorrect crawlera url: %w", err)
 	}
 
-	executor, err := httransform.MakeProxyChainExecutor(crawleraURL)
-	if err != nil {
-		return nil, fmt.Errorf("cannot make proxy chain executor: %w", err)
-	}
-
-	crawleraExecutor := func(state *httransform.LayerState) {
-		startTime := time.Now()
-
-		executor(state)
-		statsContainer.NewCrawleraTime(time.Since(startTime))
-		statsContainer.NewCrawleraRequest()
-	}
+	dialer, err := dialers.DialerFromURL(dialers.Opts{}, conf.CrawleraURL())
+	exe := executor.MakeDefaultExecutor(dialer)
+	pass, _ := crawleraURL.User.Password()
 
 	opts := httransform.ServerOpts{
-		CertCA:           []byte(conf.TLSCaCertificate),
-		CertKey:          []byte(conf.TLSPrivateKey),
-		Executor:         crawleraExecutor,
-		Logger:           &Logger{},
-		Metrics:          statsContainer,
-		Layers:           makeProxyLayers(conf, crawleraExecutor, statsContainer),
-		OrganizationName: "ScrapingHub",
+		Layers:        makeProxyLayers(conf, exe, statsContainer),
+		Executor:      exe,
+		TLSCertCA:     []byte(conf.TLSCaCertificate),
+		TLSPrivateKey: []byte(conf.TLSPrivateKey),
+		Authenticator: auth.NewBasicAuth(map[string]string{
+			crawleraURL.User.Username(): pass,
+		}),
 	}
-	if conf.Debug {
+	/*if conf.Debug {
 		opts.TracerPool = httransform.NewTracerPool(func() httransform.Tracer {
 			return &httransform.LogTracer{}
 		})
-	}
+	}*/
```

Comment:
I'm actually using debug flag and I find it useful, can we bring it back?

Reply:
Yep, it has to be brought back, that's why I commented it out instead of deleting. The problem is httransform/v2 has no word `tracer` in it, so we have to find out how to do that.

Reply:
I have removed tracer. It was actually quite confusing. I would advice to achive the same results with layers. You can add these metrics into a context and log them afterwards.

Reply:
Most of debug messages are still there. Added a couple of more.